collection/spec/collection.go

```diff
@@ -344,9 +344,9 @@ func AssertCollectionEventListenerSpec(t *testing.T, constructor Constructor) {
 		}
 
 		muppetTweetsInitID := "muppetTweetsInitID"
-		muppetTweetsName := "muppet_tweets"
+		// muppetTweetsName := "muppet_tweets"
 
-		t.Skip("TODO (b5): need user-scoped events to make this work")
+		// t.Skip("TODO (b5): need user-scoped events to make this work")
```

Comment:
not sure which events are intended to be user scoped to work, and not sure what the next lines (351-361) are supposed to be checking.

Ultimately, scoped the context to `missPiggy` for use in all subsequent events, but again, not fully sure what the intended usage should be. Should all events be ensuring correct scope in the collection, or just certain ones?

Reply:
I am also not sure what exactly the test is doing and what behavior it expects. I missed looking to this specific test when doing my recent scoped events work, but can at least provide what might be helpful info: only `ETDatasetDeleteAll` and `ETDatasetPulled` are currently using info from the context, which they obtain from the Event object creation in `publish` in event/event.go.

Perhaps others should also be doing the same thing, but I'm not totally clear on the design of collection and which events are actually intended to be global (things like "rename" probably should be?). I kept the scope of my recent work small and only modified handler that explicitly had TODOs to use scoped user info from context. Except `pushed` because that is (1) actually on the receiving end and therefore is not even using user-specific code paths and (2) not currently in use anywhere so not testable.

Reply:
great question. The thing we're trying to test: when miss piggy pulls a dataset, It shouldn't end up in Kermit's list of datasets.